cloud/amazon/ec2_vpc.py

```diff
@@ -248,14 +248,32 @@ def routes_match(rt_list=None, rt=None, igw=None):
         if route['gw'] == 'igw':
             route_kwargs['gateway_id'] = igw.id
             route_kwargs['instance_id'] = None
+            route_kwargs['interface_id'] = None
+            route_kwargs['vpc_peering_connection_id'] = None
             route_kwargs['state'] = 'active'
         elif route['gw'].startswith('i-'):
+            route_kwargs['gateway_id'] = None
             route_kwargs['instance_id'] = route['gw']
+            route_kwargs['interface_id'] = None
+            route_kwargs['vpc_peering_connection_id'] = None
+            route_kwargs['state'] = 'active'
+        elif route['gw'].startswith('eni-'):
+            route_kwargs['gateway_id'] = None
+            route_kwargs['instance_id'] = None
+            route_kwargs['interface_id'] = route['gw']
+            route_kwargs['vpc_peering_connection_id'] = None
+            route_kwargs['state'] = 'active'
+        elif route['gw'].startswith('pcx-'):
             route_kwargs['gateway_id'] = None
+            route_kwargs['instance_id'] = None
+            route_kwargs['interface_id'] = None
+            route_kwargs['vpc_peering_connection_id'] = route['gw']
             route_kwargs['state'] = 'active'
         else:
             route_kwargs['gateway_id'] = route['gw']
             route_kwargs['instance_id'] = None
+            route_kwargs['interface_id'] = None
+            route_kwargs['vpc_peering_connection_id'] = None
             route_kwargs['state'] = 'active'
         route_kwargs['destination_cidr_block'] = route['dest']
```

Comment:
For this section, could you refactor it so that at the top instead of creating route_kwargs as an empty dict it creates route_kwargs as a dict with all of these fields set to None?  Then the if-else block can just set the specific fields that it needs to and the other fields will already have been populated.